app/controllers/news_controller.rb

```diff
@@ -5,14 +5,16 @@ def index
   end
 
   def new
+    @building = Building.find(session[:current_building_id])
     @news = News.new
   end
 
   def create
     @news = News.new news_params
+    @news.update(building_id: session[:current_building_id])
```

Comment:
Hmm.. Is this passing in tests. I don't think you can update an object unless it is saved in the db. Since you create `@news` with `.new` it is NOT persistent. Not sure about this. You might be getting a false positive. I don't have the time to test this code myself so I would suggest you double check this, okay?

Reply:
Det ser ut att funka om jag inte tar fel :-/
testade en pry efter save.


From: /Users/casper/Projects/building_blocks_back_end/app/controllers/buildings_controller.rb @ line 20 BuildingsController#create:

    15: def create
    16:    @building = Building.new building_params
    17:    @building.update(user_id: current_user.id)
    18:    if @building.save
    19:      binding.pry
 
[1] pry(#<BuildingsController>)> Building.all
=> [#<Building:0x007f7f83624648
  id: 1039,
  name: "Bighouse",
  street: nil,
  zip_code: nil,
  city: nil,
  country: nil,
  created_at: Fri, 25 Aug 2017 13:31:05 UTC +00:00,
  updated_at: Fri, 25 Aug 2017 13:31:05 UTC +00:00,
  user_id: 194>]
[2] pry(#<BuildingsController>)>

Reply:
Yes, but it is saving to the database. So it is not longer `new`. And if you have any errors on the object it will raise an error and blow up the app. You should include some error handling.

Reply:
I will look into it :-)